caffe2_builders/build_windows_shared_v140.bat

```diff
@@ -0,0 +1,155 @@
+rem Remove to original folder after script is finished
+set ORIGINAL_DIR=%cd%
+
+rem Build folders are 
+rem %CAFFE2_ROOT%\build\Debug for Debug and
+rem %CAFFE2_ROOT%\build\Release for Release
+set CAFFE2_ROOT=%~dp0%\pytorch
+
+rem Cloning repository if it doesn't exist
+if not exist pytorch (
+    git clone --recursive https://github.com/ArutyunovG/pytorch.git
+    cd pytorch
+    git checkout windows-shared-build
+) else (
+    cd pytorch
+)
+
+rem Should build folder be deleted and build start from scratch?
+if NOT DEFINED CLEAR_BUILD_FOLDER_AND_START_AGAIN (
+    set CLEAR_BUILD_FOLDER_AND_START_AGAIN=0
+)
+
+rem Debug build enabled by default
+if NOT DEFINED BUILD_DEBUG (
+    set BUILD_DEBUG=1
+)
+
+rem Release build enabled by default
+if NOT DEFINED BUILD_RELEASE (
+    set BUILD_RELEASE=1
+)
+
+rem msbuild /m: option value
+if NOT DEFINED NUM_BUILD_PROC (
+    set NUM_BUILD_PROC=6
+)
+
+if %CLEAR_BUILD_FOLDER_AND_START_AGAIN% EQU 1 (
+    if exist %CAFFE2_ROOT%\build (
+        rmdir %CAFFE2_ROOT%\build /s /q
+    )
+)
+
+rem Visual Studio 14 2015 Win64 is supported by this script
+rem Define VC_BIN_ROOT where compiler and vcvars64.bat are located
+if NOT DEFINED CMAKE_GENERATOR (
+    set CMAKE_GENERATOR="Visual Studio 14 2015 Win64"
+    set VC_BIN_ROOT="%VS140COMNTOOLS%..\..\VC\bin\amd64"
+)
+
+if %VC_BIN_ROOT% EQU "" (
+    echo "Error: VC_BIN_ROOT must be specified"
+    echo "Exiting..."
+    exit
+)
+
+rem Explicitly specifiy x64 compiler to have enough heap space
+if %CMAKE_GENERATOR% EQU "Visual Studio 14 2015 Win64" (
+    set CMAKE_CXX_COMPILER=%VC_BIN_ROOT%\cl.exe
+    set CMAKE_C_COMPILER=%VC_BIN_ROOT%\cl.exe
+    set CMAKE_LINKER=%VC_BIN_ROOT%\link.exe
+)
+
+rem Now checking, that everything is defined for build...
+if %CMAKE_CXX_COMPILER% EQU "" (
+    echo "Error: CMAKE_CXX_COMPILER must be specified"
+    echo "Exiting..."
+    exit
+)
+
+if %CMAKE_C_COMPILER% EQU "" (
+    echo "Error: CMAKE_C_COMPILER must be specified"
+    echo "Exiting..."
+    exit
+)
+
+if %CMAKE_LINKER% EQU "" (
```

Comment:
Why do we want to override the default linker?

Reply:
I explicitly specify x64 toolset to build caffe2.dll in Debug with pdbs.